packages/core/src/baseclient.ts

```diff
@@ -583,7 +583,13 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
         return finalEvent.event_id;
       },
       reason => {
-        __DEBUG_BUILD__ && logger.warn(reason);
+        if (__DEBUG_BUILD__) {
+          // If something's gone wrong, log the error as a warning. If it's just us having used a `SentryError` for
+          // control flow, log just the message (no stack) as a log-level log.
+          const logLevel = (reason as SentryError).logLevel || 'warn';
+          const logee = logLevel === 'log' ? (reason as SentryError).message : reason;
+          logger[logLevel as 'log' | 'warn'](logee);
```

Comment:
Since `SentryError` is an actual class we could do:

```suggestion
          if (reason instanceof SentryError) {
            logger[reason.logLevel](reason.message);
          } else {
            logger.warn(reason);
          }
```

Note: This requires the changes in the other comment I made to work without type-casts.

Reply:
I like that. 

One complicating factor, though, is that for now `SentryError` would fail that `instanceof` test because of [this line](https://github.com/getsentry/sentry-javascript/blob/b24c7f28e4456f778799d227c21b757c1fbc0d91/packages/utils/src/error.ts#L10), though to be honest I'm not clear why we need it. (I tried taking it out, but for some reason it seems to consistently make [the ES5 playwright tests](https://github.com/getsentry/sentry-javascript/runs/7795701909?check_suite_focus=true) fail. No matter how many times I run them, the same few individual tests time out, and then that cancels all of the other matrix variations.) So not sure where to go from there.

Reply:
Oof that sucks. What is that doing? Feel free to merge as is. Not worth the effort.

Reply:
I actually did try it out, and I realized that your version (which I like a lot better than mine) actually doesn't do the same thing, unfortunately, in that it doesn't log the stack when the `SentryError` actually _is_ an error. And then once you fix that... it's more comparable to what I had. That said, I do think it's easier to read with the `if/else`, so I've restructured it that way.